test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/requirement/OpenShiftCommandLineToolsRequirement.java

```diff
@@ -18,85 +18,204 @@
 import java.nio.channels.Channels;
 import java.nio.channels.ReadableByteChannel;
 import java.nio.file.Files;
 import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Properties;
 
 import org.apache.commons.lang.StringUtils;
 import org.eclipse.core.runtime.Platform;
 import org.eclipse.osgi.util.NLS;
 import org.eclipse.reddeer.common.logging.Logger;
 import org.eclipse.reddeer.junit.requirement.Requirement;
+import org.eclipse.reddeer.workbench.ui.dialogs.WorkbenchPreferenceDialog;
 import org.jboss.tools.openshift.reddeer.exception.OpenShiftToolsException;
+import org.jboss.tools.openshift.reddeer.preference.page.OpenShift3PreferencePage;
 import org.jboss.tools.openshift.reddeer.requirement.OpenShiftCommandLineToolsRequirement.OCBinary;
+import org.jboss.tools.openshift.reddeer.utils.DatastoreOS3;
 import org.jboss.tools.openshift.reddeer.utils.FileHelper;
+import org.jboss.tools.openshift.reddeer.utils.SystemProperties;
+import org.jboss.tools.openshift.reddeer.utils.TestUtils;
 
 /**
  * Requirement to download and extract OpenShift command line tools binary which is necessary 
- * for some functionality of OpenShift tools.
+ * for some functionality of OpenShift and CDK tools. 
+ * Loads up properties file with links to different oc versions.
  *  
  * @author mlabuda@redhat.com
  * @author adietish@redhat.com
+ * @author odockal@redhat.com
  *
  */
 public class OpenShiftCommandLineToolsRequirement implements Requirement<OCBinary> {
 
 	private static final String CLIENT_TOOLS_DESTINATION = "binaries";
 	private static final String SUFFIX_TAR_GZ = ".tar.gz";
 	private static final String SUFFIX_ZIP = ".zip";
+	private WorkbenchPreferenceDialog dialog;
+	private OpenShift3PreferencePage page;
+	private static final Properties properties = TestUtils.readPropertiesFile(TestUtils.getProjectAbsolutePath("resources" + File.separator + "oclinks.properties"));
+	private String pathToOC;
+	private OCBinary binary;
 	
 	private static final Logger LOGGER = new Logger(OpenShiftCommandLineToolsRequirement.class);
 	
 	@Retention(RetentionPolicy.RUNTIME)
     @Target(ElementType.TYPE)
     public @interface OCBinary {
+		/**
+		 * Decides whether existing oc on pathToOC path will be used.
+		 * Is in exclusion with downloadNewestOC.
+		 * @return false is default value
+		 */
+		boolean ocExists() default false;
+		
+		/**
+		 * Path to existing oc, ie. one that is configured within minishift setup-cdk
+		 * @return path to existing oc
+		 */
+		String pathToOC() default StringUtils.EMPTY;
+		
+		/**
+		 * Sets if default oc version or newest will be downloaded
+		 * @return true or false
+		 */
+		boolean latestOC() default true;
+		
+		/**
+		 * If set to true, given oc will be set into 
+		 * Preferences -> JBoss Tools -> OpenShift 3 -> oc executable location field
+		 * @return true or false
+		 */
+		boolean setOCInPrefs() default false;
+		
+		/**
+		 * Will delete downloaded oc, is not used when ocExists=true
```

Comment:
isn't "cleanup=false" missing in the javadoc?

Reply:
@adietish all comments taken into account, ready to push it. Unless you are still reviewing...